server/schedulers/split_bucket.go

```diff
@@ -0,0 +1,257 @@
+	conf := h.conf.Clone()
+	_ = h.rd.JSON(w, http.StatusOK, conf)
+}
+
+func (h *splitBucketHandler) UpdateConfig(w http.ResponseWriter, r *http.Request) {
+	h.conf.mu.Lock()
+	defer h.conf.mu.Unlock()
+	rd := render.New(render.Options{IndentJSON: true})
+	oldc, _ := json.Marshal(h.conf)
+	data, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		rd.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	if err := json.Unmarshal(data, h.conf); err != nil {
+		rd.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	newc, _ := json.Marshal(h.conf)
+	if !bytes.Equal(oldc, newc) {
+		h.conf.persistLocked()
+		rd.Text(w, http.StatusOK, "success")
+	}
+
+	rd.Text(w, http.StatusBadRequest, "config item not found")
+}
+
+func newSplitBucketHandler(conf *splitBucketSchedulerConfig) http.Handler {
+	h := &splitBucketHandler{
+		conf: conf,
+		rd:   render.New(render.Options{IndentJSON: true}),
+	}
+	router := mux.NewRouter()
+	router.HandleFunc("/list", h.ListConfig).Methods("GET")
+	router.HandleFunc("/config", h.UpdateConfig).Methods("POST")
+	return router
+}
+
+func newSplitBucketScheduler(opController *schedule.OperatorController, conf *splitBucketSchedulerConfig) *splitBucketScheduler {
+	base := NewBaseScheduler(opController)
+	handler := newSplitBucketHandler(conf)
+	ret := &splitBucketScheduler{
+		BaseScheduler: base,
+		conf:          conf,
+		handler:       handler,
+	}
+	return ret
+}
+
+// GetName returns the name of the split bucket scheduler.
+func (s *splitBucketScheduler) GetName() string {
+	return SplitBucketName
+}
+
+// GetType returns the type of the split bucket scheduler.
+func (s *splitBucketScheduler) GetType() string {
+	return SplitBucketType
+}
+
+// ServerHTTP implement Http server.
+func (s *splitBucketScheduler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.handler.ServeHTTP(w, r)
+}
+
+// IsScheduleAllowed return true if the sum of executing opSplit operator is less  .
+func (s *splitBucketScheduler) IsScheduleAllowed(_ schedule.Cluster) bool {
+	allowed := s.BaseScheduler.OpController.OperatorCount(operator.OpSplit) < s.conf.SplitLimit
+	if !allowed {
+		operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpSplit.String()).Inc()
+	}
+	return allowed
+}
+
+type splitBucketPlan struct {
+	hotBuckets         map[uint64][]*buckets.BucketStat
+	cluster            schedule.Cluster
+	conf               *splitBucketSchedulerConfig
+	hotRegionSplitSize int64
+}
+
+// Schedule return operators if some bucket is too hot.
+func (s *splitBucketScheduler) Schedule(cluster schedule.Cluster) []*operator.Operator {
+	schedulerCounter.WithLabelValues(s.GetName(), "schedule").Inc()
+	conf := s.conf.Clone()
+	plan := &splitBucketPlan{
+		conf:               conf,
+		cluster:            cluster,
+		hotBuckets:         cluster.BucketsStats(conf.Degree),
+		hotRegionSplitSize: cluster.GetOpts().GetHotRegionSplitSize(),
```

Comment:
Again, it seems that this split size config is only used in the `split-bucket-scheduler`. We should use a better name to indicate it to show its usage more clearly. Otherwise, it's easy to omit its actual intrinsic connection between dynamic region and bucket.

Reply:
yes, it will comment `The threshold of region size for balance hot region scheduler and split bucket. \n
	// It will be split for move hot region if region is bigger than HotRegionSplitSize.`